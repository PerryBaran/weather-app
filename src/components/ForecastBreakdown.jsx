import React from "react";
import PropTypes from "prop-types";
import WeatherIcon from "react-icons-weather";
import useUnits from "../helpers/getUnits";
import useUnitContext from "../hooks/useUnitContext";
import "../styles/ForecastBreakdown.css";

function ForecastBreakdown({
  icon,
  temp,
  minTemp,
  maxTemp,
  humidity,
  windSpeed,
  weather,
}) {
  const { units } = useUnitContext();
  const { tempUnits, speedUnits } = useUnits(units);

  return (
    <div className="forecast-breakdown">
      <div className="forecast-breakdown__right">
        <div className="forecast-breakdown__icon" data-testid="forecast-icon">
          <WeatherIcon name="owm" iconId={icon} />
        </div>
        <p className="forecast-breakdown__weather">{weather}</p>
      </div>
      <div className="forecast-breakdown__left">
        <p className="forecast-breakdown__temp">
          {temp}
          <span>{tempUnits}</span>
        </p>
        <p className="forecast-breakdown__min-temp">
          <span>Min: </span>
          {minTemp}
          {tempUnits}
        </p>
        <p className="forecast-breakdown__max-temp">
          <span>Max: </span>
          {maxTemp}
          {tempUnits}
        </p>
        <p className="forecast-breakdown__humidity">
          <span>Humidity: </span>
          {humidity}%
        </p>
        <p className="forecast-breakdown__wind-speed">
          <span>Wind Speed: </span>
          {windSpeed}
          {speedUnits}
        </p>
      </div>
    </div>
  );
}

ForecastBreakdown.propTypes = {
  icon: PropTypes.string.isRequired,
  temp: PropTypes.string.isRequired,
  minTemp: PropTypes.string.isRequired,
  maxTemp: PropTypes.string.isRequired,
  humidity: PropTypes.string.isRequired,
  windSpeed: PropTypes.string.isRequired,
  weather: PropTypes.string.isRequired,
};

export default ForecastBreakdown;
