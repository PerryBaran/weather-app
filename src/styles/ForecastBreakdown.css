.forecast-breakdown {
  margin-top: 20px;
  text-align: center;
}

.forecast-breakdown span {
  font-weight: 600;
}