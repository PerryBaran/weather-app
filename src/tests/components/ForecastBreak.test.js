import React from "react";
import { render } from "@testing-library/react";
import ForecastBreakdown from "../../components/ForecastBreakdown";

describe("ForecastBreakdown", () => {
  const validProps = {
    temp: "20",
    minTemp: "18",
    maxTemp: "22",
    humidity: "30",
    windSpeed: "40",
  };

  describe("snapshot", () => {
    it("renders correctly", () => {
      const { asFragment } = render(
        <ForecastBreakdown
          temp={validProps.temp}
          minTemp={validProps.minTemp}
          maxTemp={validProps.maxTemp}
          humidity={validProps.humidity}
          windSpeed={validProps.windSpeed}
        />
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("props", () => {
    it("renders correct values for props", () => {
      const { getByText } = render(
        <ForecastBreakdown
          temp={validProps.temp}
          minTemp={validProps.minTemp}
          maxTemp={validProps.maxTemp}
          humidity={validProps.humidity}
          windSpeed={validProps.windSpeed}
        />
      );

      expect(getByText(`${validProps.temp}°C`)).toHaveAttribute(
        "class",
        "forecast-breakdown__temp"
      );
      expect(getByText(`${validProps.minTemp}°C`)).toHaveAttribute(
        "class",
        "forecast-breakdown__min-temp"
      );
      expect(getByText(`${validProps.maxTemp}°C`)).toHaveAttribute(
        "class",
        "forecast-breakdown__max-temp"
      );
      expect(getByText(`${validProps.humidity}%`)).toHaveAttribute(
        "class",
        "forecast-breakdown__humidity"
      );
      expect(getByText(`${validProps.windSpeed}km/h`)).toHaveAttribute(
        "class",
        "forecast-breakdown__wind-speed"
      );
    });
  });
});
