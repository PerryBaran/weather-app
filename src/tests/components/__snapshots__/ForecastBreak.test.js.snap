// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastBreakdown snapshot renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-breakdown"
  >
    <p
      class="forecast-breakdown__temp"
    >
      <span>
        Temperture: 
      </span>
      20°C
    </p>
    <p
      class="forecast-breakdown__min-temp"
    >
      <span>
        Minimum Temperature:
      </span>
      18°C
    </p>
    <p
      class="forecast-breakdown__max-temp"
    >
      <span>
        Maximum Temperature:
      </span>
      22°C
    </p>
    <p
      class="forecast-breakdown__humidity"
    >
      <span>
        Humidity:
      </span>
      30%
    </p>
    <p
      class="forecast-breakdown__wind-speed"
    >
      <span>
        Wind Speed:
      </span>
      40km/h
    </p>
  </div>
</DocumentFragment>
`;
